File: src/main/java/learn/script/mapper/userMapper.js
/**
 * Created by flysLi on 2018/12/17.
 */
var sql = "select * from user where 1=1 ";

function selectAll(map){
    if(map.name!=null){
        sql+=" and name='"+map.name+"'"
    }
    if(map.name!=null){
        sql+=" and age='"+map.age+"'"
    }
    if(map.name!=null){
        sql+=" and address='"+map.address+"'"
    }
    if(map.name!=null){
        sql+=" and bird='"+map.bird+"'"
    }
    return sql;
}

function showName(name) {
    return "SELECT * FROM USER WHERE NAME='"+name+"';";
}

File: src/main/java/learn/script/dao/UserMapper.java
/*
 * Copyright (c) 2018, 2018, Travel and/or its affiliates. All rights reserved.
 * TRAVEL PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 */
package script.dao;

import script.common.annotations.Drop;
import script.common.annotations.NameSpace;
import script.core.BaseMapper;
import script.entity.User;

/**
 * @author flysLi
 * @ClassName UserMapper
 * @Decription TODO
 * @Date 2018/12/17 14:04
 * @Version 1.0
 */
@NameSpace(path = "D:\\workspace_srudent\\comflyseureka\\src\\main\\java\\script\\mapper\\userMapper.js")
public interface UserMapper extends BaseMapper<User, Integer> {

    @Drop(type = "Select")
    User showName(String name);

    @Drop(type = "Select")
    User findOne(int id);
}
